dwave/cloud/cli.py

```diff
@@ -950,20 +976,32 @@ def auth():
 @click.option('--oob', is_flag=True,
               help='Run OAuth 2.0 Authorization Code flow out-of-band, '
                    'without the use of locally hosted redirect URL.')
+@click.option('--skip-valid', is_flag=True,
+              help='Skip authorization if access token is valid, or it '
+                   'can be refreshed.')
 @standardized_output
-def login(*, config_file, profile, oob, output):
+def login(*, config_file, profile, oob, skip_valid, output):
     """Authorize Ocean to access Leap API on user's behalf."""
 
+    return _login(config_file=config_file, profile=profile, oob=oob,
+                  skip_valid=skip_valid, output=output)
+
+def _login(*, config_file, profile, oob, skip_valid, output):
     config = validate_config_v1(load_config(config_file=config_file, profile=profile))
     flow = LeapAuthFlow.from_config_model(config)
 
+    if skip_valid:
+        if flow.ensure_active_token():
+            output('Valid token found, skipping authorization.')
+            return
+
     if oob:
         flow.run_oob_flow(open_browser=True)
     else:
         flow.run_redirect_flow(open_browser=True)
 
-    click.echo('Authorization completed successfully. '
-               'You can now use "dwave auth get" to fetch your token.')
+    output('Authorization completed successfully. '
```

Comment:
I am a bit confused about where we use `output()` vs `click.echo()`. For instance in `setup()` they are all `click.echo()`.

Reply:
Something we need to clean-up. `output` is provided by `@standardized_output`, and it handles plain text output and JSON output. But `setup` predates all that, so needs updating. I probably missed it in other places as well, like in login.

Generally, it doesn't make much sense to use it in interactive commands, like auth login is.